core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1634,71 +1674,229 @@ boolean resumeSyncForDocument(
+    for (final BsonDocument document : documents) {
+      final BsonValue documentId = BsonUtils.getDocumentId(document);
+      final ChangeEvent<BsonDocument> event = changeEventForLocalInsert(namespace, document, true);
+      final CoreDocumentSynchronizationConfig config = syncConfig.addSynchronizedDocument(
+          namespace,
+          documentId
+      );
+      config.setSomePendingWrites(logicalT, event);
+      emitEvent(documentId, event);
     }
+    triggerListeningToNamespace(namespace);
+  }
 
-    // TODO: STITCH-1958
-    final BsonDocument documentBeforeUpdate =
-        getLocalCollection(namespace).find(getDocumentIdFilter(documentId)).first();
+  /**
+   * Update a single document in the collection according to the specified arguments.
+   *
+   * @param filter a document describing the query filter, which may not be null.
+   * @param update a document describing the update, which may not be null. The update to
+   *               apply must include only update operators.
+   * @return the result of the update one operation
+   */
+  UpdateResult updateOne(final MongoNamespace namespace, final Bson filter, final Bson update) {
+    return updateOne(namespace, filter, update, new UpdateOptions());
+  }
 
-    final BsonDocument result = getLocalCollection(namespace)
-        .findOneAndUpdate(
-            getDocumentIdFilter(documentId),
-            update,
-            new FindOneAndUpdateOptions().returnDocument(ReturnDocument.AFTER));
+  /**
+   * Update a single document in the collection according to the specified arguments.
+   *
+   * @param filter        a document describing the query filter, which may not be null.
+   * @param update        a document describing the update, which may not be null. The update to
+   *                      apply must include only update operators.
+   * @param updateOptions the options to apply to the update operation
+   * @return the result of the update one operation
+   */
+  UpdateResult updateOne(
+      final MongoNamespace namespace,
+      final Bson filter,
+      final Bson update,
+      final UpdateOptions updateOptions) {
+    // read the local collection
+    final MongoCollection<BsonDocument> localCollection = getLocalCollection(namespace);
+
+    // fetch the document prior to updating
+    final BsonDocument documentBeforeUpdate = getLocalCollection(namespace).find(filter).first();
+
+    // if there was no document prior and this is not an upsert,
+    // do not acknowledge the update
+    if (!updateOptions.isUpsert() && documentBeforeUpdate == null) {
+      return UpdateResult.acknowledged(0, 0L, null);
+    }
 
-    if (result == null) {
+    // find and update the single document, returning the document post-update
+    final BsonDocument documentAfterUpdate = localCollection.findOneAndUpdate(
+        filter,
+        update,
+        new FindOneAndUpdateOptions()
+            .collation(updateOptions.getCollation())
```

Comment:
[minor] collation, bypassDocumentValidation, and arrayFilters are all unsupported by the remote MongoDB service so this might result in conflicts between local and remote where a document that is able to inserted locally is not able to be inserted remotely. we could either not support those options at all, or just do this with the assumption that we will eventually support the full mongodb surface area

Reply:
There's no actual way for these to get passed downstream, if you look through the proxies– we do not actually support the options. But we seem to operate within the DataSynchronizer on localmdb without considering Remote.